scripts/zkai_d64_native_relation_witness_oracle.py

```
@@ -0,0 +1,606 @@
+    try:
+        timestamp = int(raw)
+    except ValueError as err:
+        raise NativeRelationWitnessOracleError("SOURCE_DATE_EPOCH must be an integer timestamp") from err
+    try:
+        generated_at = dt.datetime.fromtimestamp(timestamp, tz=dt.timezone.utc)
+    except (OverflowError, OSError, ValueError) as err:
+        raise NativeRelationWitnessOracleError("SOURCE_DATE_EPOCH must be in the supported timestamp range") from err
+    return generated_at.isoformat().replace("+00:00", "Z")
+
+
+def _validate_generated_at(value: Any) -> None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)